File: app/src/main/java/com/bykea/pk/partner/ui/activities/ProblemActivity.java
package com.bykea.pk.partner.ui.activities;

import android.os.Bundle;
import android.view.WindowManager;
import android.widget.FrameLayout;
import android.widget.ImageView;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.bykea.pk.partner.R;
import com.bykea.pk.partner.ui.fragments.ProblemListFragment;
import com.bykea.pk.partner.ui.fragments.ProblemSubmittedFragment;
import com.bykea.pk.partner.widgets.FontTextView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ProblemActivity extends BaseActivity {

    private ProblemActivity mCurrentActivity;
    public String tripId;
    public String selectedReason;

    public static final String LIST_FRAGMENT = "LIST_FRAGMENT";
    public static final String DETAIL_FRAGMENT = "DETAIL_FRAGMENT";
    public static final String DETAIL_SUBMITTED_FRAGMENT = "DETAIL_SUBMITTED_FRAGMENT";

    @BindView(R.id.toolbar)
    FrameLayout toolbar;
    @BindView(R.id.ivBackBtn)
    ImageView ivBackBtn;
    @BindView(R.id.tvTitle)
    FontTextView tvTitle;
    private FragmentManager fragmentManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_problem);
        overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        mCurrentActivity = this;
        ButterKnife.bind(mCurrentActivity);

        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
        tripId = getIntent().getStringExtra("TRIP_ID");
        fragmentManager = getSupportFragmentManager();
        loadFragment(new ProblemListFragment(), LIST_FRAGMENT);
        setGreenActionBarTitle(tripId, "");
    }

    public void loadFragment(Fragment fragment, String fragmentTag) {
        fragmentManager
                .beginTransaction()
                .replace(R.id.containerView, fragment, fragmentTag)
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
                .commit();
    }

    @Override
    public void onBackPressed() {
        /*switch (getSupportFragmentManager().getFragments()
                .get(1).getTag()) {

        }*/
        if (getSupportFragmentManager().findFragmentByTag(DETAIL_SUBMITTED_FRAGMENT) instanceof ProblemSubmittedFragment) {
            finish();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/com/bykea/pk/partner/ui/support/ProblemActivity.kt
package com.bykea.pk.partner.ui.support

import android.os.Bundle
import android.view.MenuItem
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.bykea.pk.partner.R
import com.bykea.pk.partner.databinding.ActivityProblemBinding
import com.bykea.pk.partner.ui.loadboard.common.obtainViewModel

import kotlinx.android.synthetic.main.activity_problem.*


class ProblemActivity : AppCompatActivity() {

    private lateinit var binding: ActivityProblemBinding
    private lateinit var mCurrentActivity: ProblemActivity
    private var fragmentManager: FragmentManager? = null

    private var tripId: String? = null
    var selectedReason: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_problem)
        mCurrentActivity = this

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)

        tripId = intent.getStringExtra("TRIP_ID")
        fragmentManager = supportFragmentManager
        changeFragment(ProblemListFragment(), LIST_FRAGMENT)

        binding.viewmodel = obtainViewModel(ProblemViewModel::class.java)
        binding.listener = object : ProblemListener {

        }
    }

    fun changeFragment(fragment: Fragment, fragmentTag: String) {
        fragmentManager!!
                .beginTransaction()
                .replace(R.id.containerView, fragment, fragmentTag)
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
                .commit()
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        val id = item?.getItemId()

        if (id == android.R.id.home) {
            onBackPressed()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        super.onBackPressed()
    }

    companion object {
        val DETAIL_FRAGMENT: String = "DETAIL_FRAGMENT"
        val LIST_FRAGMENT: String = "LIST_FRAGMENT"
    }
}
